_pytest/assertion/rewrite.py

```diff
@@ -35,6 +35,12 @@
 REWRITE_NEWLINES = sys.version_info[:2] != (2, 7) and sys.version_info < (3, 2)
 ASCII_IS_DEFAULT_ENCODING = sys.version_info[0] < 3
 
+if sys.version_info >= (3,5):
```

Comment:
I looked into this (I'm collaborating with @Carreau now on the PR), and I don't think there's a good way of doing that. The `ast.Call` type does not have an introspectable signature, so the only way to check how many arguments it takes is to try constructing it, and catch TypeError. That's doable, but if e.g. Python 3.6 added two new arguments again, it wouldn't distinguish between the new five-argument form and the old one.

So I think it is actually better to switch based on the Python version in this case.